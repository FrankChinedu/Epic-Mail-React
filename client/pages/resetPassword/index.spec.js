import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import ResetPassword, { ResetPasswordComponent } from './index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
let store;

describe('Reset Password', () => {
  beforeEach(() => {
    store = mockStore({
      auth: {
        authError: null,
        user: []
      }
    });
  });
  it('should render reset password page', () => {
    const history = {
      history: {
        location: {
          search: '?x-access-token=snvdhrddcvcbhhfnfbfbbvbfv'
        }
      }
    };
    const component = shallow(<ResetPasswordComponent history={history} />);

    expect(component).toMatchSnapshot();
  });

  it('should render reset password with store', () => {
    const history = {
      history: {
        location: {
          search: 'somethind'
        }
      }
    };
    const component = mount(
      <Router>
        <ResetPassword store={store} history={history} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });
});
