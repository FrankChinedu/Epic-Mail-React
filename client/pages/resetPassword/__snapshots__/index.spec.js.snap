// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password should render reset password page 1`] = `
<Fragment>
  <main
    className="flex justify-content-ctr mt-55"
  >
    <div
      className="col-5 mt-5 form-cover"
    >
      <h3
        className="center-text"
      >
        Reset Password
      </h3>
      <Formik
        enableReinitialize={false}
        initialValues={
          Object {
            "password": "",
            "passwordConfirmation": "",
          }
        }
        isInitialValid={false}
        onSubmit={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
        validationSchema={
          ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_nodes": Array [
              "passwordConfirmation",
              "password",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "password": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "min": true,
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "passwordConfirmation": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {
                    null,
                  },
                  "refs": Map {
                    "password" => Reference {
                      "getter": [Function],
                      "isContext": false,
                      "isSibling": true,
                      "isValue": false,
                      "key": "password",
                      "map": undefined,
                      "path": "password",
                    },
                  },
                },
                "_whitelistError": [Function],
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
          }
        }
      >
        <Component />
      </Formik>
    </div>
  </main>
</Fragment>
`;

exports[`Reset Password should render reset password with store 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(ResetPassword)
      history={
        Object {
          "history": Object {
            "location": Object {
              "search": "somethind",
            },
          },
        }
      }
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <ResetPassword
        history={
          Object {
            "history": Object {
              "location": Object {
                "search": "somethind",
              },
            },
          }
        }
        onSubmit={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      >
        <main
          className="flex justify-content-ctr mt-55"
        >
          <div
            className="col-5 mt-5 form-cover"
          >
            <h3
              className="center-text"
            >
              Reset Password
            </h3>
            <Formik
              enableReinitialize={false}
              initialValues={
                Object {
                  "password": "",
                  "passwordConfirmation": "",
                }
              }
              isInitialValid={false}
              onSubmit={[Function]}
              validateOnBlur={true}
              validateOnChange={true}
              validationSchema={
                ObjectSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": undefined,
                  "_nodes": Array [
                    "passwordConfirmation",
                    "password",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "fields": Object {
                    "password": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "min": true,
                        "required": true,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "passwordConfirmation": StringSchema {
                      "_blacklist": RefSet {
                        "list": Set {},
                        "refs": Map {},
                      },
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": true,
                      },
                      "_mutate": undefined,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": RefSet {
                        "list": Set {
                          null,
                        },
                        "refs": Map {
                          "password" => Reference {
                            "getter": [Function],
                            "isContext": false,
                            "isSibling": true,
                            "isValue": false,
                            "key": "password",
                            "map": undefined,
                            "path": "password",
                          },
                        },
                      },
                      "_whitelistError": [Function],
                      "tests": Array [
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                }
              }
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <input
                    autoComplete="true"
                    className="form-control mt-5"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
                  />
                  <p
                    className="center-text"
                  />
                </div>
                <div
                  className="mt-25"
                >
                  <label
                    htmlFor="passwordConfirmation"
                  >
                    Confirm Password
                  </label>
                  <input
                    autoComplete="true"
                    className="form-control mt-5"
                    name="passwordConfirmation"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="confirm password"
                    type="password"
                    value=""
                  />
                  <p
                    className="center-text"
                  />
                </div>
                <div
                  className="center-text mt-25"
                />
                <div
                  className="mt-25 center-text"
                >
                  <button
                    className="mt-25 btn"
                    disabled={false}
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </Formik>
          </div>
        </main>
      </ResetPassword>
    </Connect(ResetPassword)>
  </Router>
</BrowserRouter>
`;
